jwst/cube_build/blot_cube_build.py

```diff
@@ -131,98 +131,144 @@ def blot_images(self):
-                blot.meta.filename = filename[:indx] + '_blot.fits' #set output name
-                ydet,xdet=np.mgrid[:2048,:2048]
-
-# determine the ra and dec values for each x,y pixel on the detector.
-
+                blot.meta.filename = filename[:indx] + '_blot.fits' 
+                ra_det=np.zeros((2048,2048))
+                dec_det=np.zeros((2048,2048))
+                lam_det=np.zeros((2048,2048))
+                flag_det = np.zeros((2048,2048))
+                
+                # for NIRSPEC each file has 30 slices - 
+                # wcs information access seperately for each slice
+                start_slice = 0
+                end_slice = 29
+                nslices = end_slice - start_slice + 1
+                regions = list(range(start_slice, end_slice + 1))
+#                log.info('Looping over 30 slices on NIRSPEC detector, this takes a little while')
+                for ii in regions:
+                    slice_wcs = nirspec.nrs_wcs_set_input(model, ii)
+                    x,y = wcstools.grid_from_bounding_box(slice_wcs.bounding_box)
+                    ra, dec, lam = slice_wcs(x, y)
+
+                    valid1 = np.isfinite(ra)
+                    valid2 = np.isfinite(dec)
+                    valid3 = np.isfinite(lam)
+                    value = valid1 & valid2 & valid3
+                    good_data =  np.where( value == True)
+                    
+                    ra = ra[good_data]
+                    dec = dec[good_data]
+                    lam = lam[good_data]
+                    x = x[good_data]
+                    y = y[good_data]
+                    
+                    xind = x.astype(np.int)
+                    yind = y.astype(np.int)
+                    xind = np.ndarray.flatten(xind)
+                    yind = np.ndarray.flatten(yind)
+                    ra = np.ndarray.flatten(ra)
+                    dec = np.ndarray.flatten(dec)
+                    lam = np.ndarray.flatten(lam)
+                    ra_det[yind,xind] = ra 
+                    dec_det[yind,xind] = dec
+                    lam_det[yind,xind] = lam
+                    flag_det[yind,xind] = ra*0.0 + 1
+# Done looping over slices 
             log.info('Blotting back %s',model.meta.filename)
-            ra_det,dec_det,wave_det = model.meta.wcs(xdet,ydet)
 
-            valid1 = np.isfinite(ra_det)
-            valid2 = np.isfinite(dec_det)
-            valid3 = np.isfinite(wave_det)
             if self.instrument == 'MIRI':
+                valid1 = np.isfinite(ra_det)
+                valid2 = np.isfinite(dec_det)
+                valid3 = np.isfinite(lam_det)
                 value = valid1 & valid2 & valid3 & pixel_mask
-            elif elf.instrument == 'NIRSPEC':
-                value = valid1 & valid2 & valid3
+                good_data =  np.where( value == True)
+            elif self.instrument == 'NIRSPEC':
+                good_data = np.where(flag_det == 1)
```

Comment:
Are the Nirspec specific lines still needed or is this handled up in the file?

Reply:
I first set ra_det, dec_det,lam_det = np.zeros((2048,2048))
I then loop over the slices and filling the valid pixels with good data.
In gap slices I have 0 values for ra_det,dec_det,lam_det so I carry the flag_det array to
check when there is a gap